src/ado2gh/Commands/DownloadLogsCommand.cs

```diff
@@ -11,11 +11,15 @@ namespace OctoshiftCLI.AdoToGithub.Commands
 {
     public class DownloadLogsCommand : Command
     {
+        internal const int WaitIntervalInSeconds = 10;
+        internal const int RetryMaxCount = 6;
+
         private readonly OctoLogger _log;
         private readonly GithubApiFactory _githubApiFactory;
         private readonly HttpDownloadService _httpDownloadService;
 
         internal Func<string, bool> FileExists = path => File.Exists(path);
+        internal Func<int, Task> Delay = milliSeconds => Task.Delay(milliSeconds);
```

Comment:
Why do we need to abstract out the `Delay`? For testing we can set `WaitIntervalInSeconds` and `RetryMaxCount`

Reply:
Also as a [best practice](https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task) the delegate has to be async and the `Task.Delay` should be awaited:
```c#
internal Func<int, Task> Delay = async milliSeconds => await Task.Delay(milliSeconds);
```